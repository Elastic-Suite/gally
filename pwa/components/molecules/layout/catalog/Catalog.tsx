import { Title } from '~/components/atoms/title/Title'
import CatalogTabs from '~/components/atoms/catalog/catalogTabs/CatalogTabs'

const Catalog = () => {
  return (
    <>
      <Title name="Catalog" />
      <CatalogTabs />
    </>
  )
}

export default Catalog
