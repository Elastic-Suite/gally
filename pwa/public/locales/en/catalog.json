{
  "catalog_one": "catalog",
  "catalog_other": "catalogs",
  "activeLocale_one": "active locale",
  "activeLocale_other": "active locales"
}
