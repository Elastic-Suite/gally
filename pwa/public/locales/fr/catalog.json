{
  "title.catalogs": "Catalogs",
  "title.activeLocales": "Langues activ√©es"
}
