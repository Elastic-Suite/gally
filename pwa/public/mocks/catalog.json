{
  "@context": "/contexts/Catalog",
  "@id": "/catalogs",
  "@type": "hydra:Collection",
  "hydra:member": [
    {
      "@id": "/catalogs/1",
      "@type": "Catalog",
      "id": 1,
      "code": "com",
      "name": "COM Catalog",
      "localizedCatalogs": [
        {
          "@id": "/localized_catalogs/1",
          "@type": "LocalizedCatalog",
          "id": 1,
          "name": "COM French Catalog",
          "code": "com_fr",
          "locale": "fr_FR",
          "isDefault": false,
          "localName": "testUn (France)"
        },
        {
          "@id": "/localized_catalogs/1",
          "@type": "LocalizedCatalog",
          "id": 1,
          "name": "COM French Catalog",
          "code": "com_fr",
          "locale": "fr_FR",
          "isDefault": false,
          "localName": "testUnAA (France)"
        },
        {
          "@id": "/localized_catalogs/1",
          "@type": "LocalizedCatalog",
          "id": 1,
          "name": "COM French Catalog",
          "code": "com_fr",
          "locale": "fr_FR",
          "isDefault": false,
          "localName": "testAAUn (France)"
        },
        {
          "@id": "/localized_catalogs/1",
          "@type": "LocalizedCatalog",
          "id": 1,
          "name": "COM French Catalog",
          "code": "com_fr",
          "locale": "fr_FR",
          "isDefault": false,
          "localName": "tAAestUn (France)"
        },
        {
          "@id": "/localized_catalogs/1",
          "@type": "LocalizedCatalog",
          "id": 1,
          "name": "COM French Catalog",
          "code": "com_fr",
          "locale": "fr_FR",
          "isDefault": false,
          "localName": "tesAEFtUn (France)"
        },
        {
          "@id": "/localized_catalogs/1",
          "@type": "LocalizedCatalog",
          "id": 1,
          "name": "COM French Catalog",
          "code": "com_fr",
          "locale": "fr_FR",
          "isDefault": false,
          "localName": "tesRDGFCDeux (France)"
        },
        {
          "@id": "/localized_catalogs/2",
          "@type": "LocalizedCatalog",
          "id": 2,
          "name": "COM English Catalog",
          "code": "com_en",
          "locale": "en_US",
          "isDefault": false,
          "localName": "deutGFDsh (États-Unis)"
        },
        {
          "@id": "/localaized_catalogs/2",
          "@type": "LocaalizedCatalog",
          "id": 9,
          "name": "COM Enaglish Catalog",
          "code": "com_aen",
          "locale": "en_aUS",
          "isDefault": false,
          "localName": "deutGFDsh (États-Unis)"
        }
      ]
    },
    {
      "@id": "/catalogs/2",
      "@type": "Catalog",
      "id": 2,
      "code": "fr",
      "name": "FR Catalog",
      "localizedCatalogs": [
        {
          "@id": "/localized_catalogs/3",
          "@type": "LocalizedCatalog",
          "id": 3,
          "name": "FR French Catalog",
          "code": "fr_fr",
          "locale": "fr_FR",
          "isDefault": false,
          "localName": "français (France)"
        },
        {
          "@id": "/localized_catalogs/4",
          "@type": "LocalizedCatalog",
          "id": 4,
          "name": "FR English Catalog",
          "code": "fr_en",
          "locale": "en_US",
          "isDefault": false,
          "localName": "anglais (États-Unis)"
        },
        {
          "@id": "/localized_catalogs/1",
          "@type": "LocalizedCatalog",
          "id": 1,
          "name": "COM French Catalog",
          "code": "com_fr",
          "locale": "fr_FR",
          "isDefault": false,
          "localName": "tAAestUn (France)"
        },
        {
          "@id": "/localized_catalogs/1",
          "@type": "LocalizedCatalog",
          "id": 1,
          "name": "COM French Catalog",
          "code": "com_fr",
          "locale": "fr_FR",
          "isDefault": false,
          "localName": "tesAEFtUn (France)"
        },
        {
          "@id": "/localized_catalogs/1",
          "@type": "LocalizedCatalog",
          "id": 1,
          "name": "COM French Catalog",
          "code": "com_fr",
          "locale": "fr_FR",
          "isDefault": false,
          "localName": "tesRDGFCDeux (France)"
        },
        {
          "@id": "/localized_catalogs/2",
          "@type": "LocalizedCatalog",
          "id": 2,
          "name": "COM English Catalog",
          "code": "com_en",
          "locale": "en_US",
          "isDefault": false,
          "localName": "deutGFDsh (États-Unis)"
        }
      ]
    },
    {
      "@id": "/catalogs/3",
      "@type": "Catalog",
      "id": 3,
      "code": "uk",
      "name": "UK Catalog",
      "localizedCatalogs": [
        {
          "@id": "/localized_catalogs/5",
          "@type": "LocalizedCatalog",
          "id": 5,
          "name": "EN French Catalog",
          "code": "en_fr",
          "locale": "fr_FR",
          "isDefault": false,
          "localName": "français (France)"
        }
      ]
    }
  ],
  "hydra:totalItems": 3,
  "hydra:view": {
    "@id": "/catalogs?pageSize=30",
    "@type": "hydra:PartialCollectionView"
  }
}
