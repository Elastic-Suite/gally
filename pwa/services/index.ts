export * from './api'
export * from './authProvider'
export * from './format'
export * from './hydra'
export * from './jsonld'
export * from './local'
export * from './table'
export * from './tests'
