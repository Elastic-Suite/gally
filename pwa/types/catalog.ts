import { IHydraMember, IJsonldId, IJsonldType } from './hydra'

export interface ILocalizedCatalog extends IJsonldType, IJsonldId {
  id: number
  name: string
  code: string
  locale: string
  isDefault: boolean
  localName: string
}

export interface ICatalog extends IHydraMember {
  name: string
  localizedCatalogs: ILocalizedCatalog[]
}
