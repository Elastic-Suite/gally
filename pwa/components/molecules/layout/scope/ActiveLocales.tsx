import { styled } from '@mui/material/styles'
import TitleScope from '~/components/atoms/scope/TitleScope'
import NbActiveLocales from '~/components/atoms/scope/NbActiveLocales'
import Language from '~/components/atoms/scope/Language'
import { IHydraResponse } from '~/types'

const CustomRoot = styled('div')(({ theme }) => ({
  width: '671px',
  boxSizing: 'border-box',
  display: 'flex',
  flexDirection: 'column',
  gap: theme.spacing(3),
  padding: theme.spacing(4),
  background: theme.palette.colors.white,
  border: '1px solid #E2E6F3',
  borderRadius: 8,
}))

interface IProps {
  content: IHydraResponse | IHydraResponse[]
}

function ActiveLocales({ content }: IProps): JSX.Element {
  let Languages = []
  for (const hydraContent of content['hydra:member']) {
    for (const localizedCatalogsContent of hydraContent.localizedCatalogs) {
      Languages = [...Languages, localizedCatalogsContent.localName]
    }
  }
  Languages = [...new Set(Languages)]

  return (
    <CustomRoot>
      <TitleScope name="Total" />
      <NbActiveLocales number={Languages.length} />
      <Language language={Languages} limit={false} />
    </CustomRoot>
  )
}

export default ActiveLocales
