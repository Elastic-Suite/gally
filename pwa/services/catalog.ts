import { ICatalog } from '~/types'

export function findDefaultCatalog(catalogsData: ICatalog[]): ICatalog | null {
  const defaultCatalog = catalogsData
    ? catalogsData.filter((catalog) =>
        catalog.localizedCatalogs.find((localizedCtl) => localizedCtl.isDefault)
      )[0]
    : null

  if (defaultCatalog) {
    defaultCatalog.localizedCatalogs = defaultCatalog.localizedCatalogs.filter(
      (localizedCtl) => localizedCtl.isDefault
    )
  }
  return defaultCatalog ? defaultCatalog : null
}
