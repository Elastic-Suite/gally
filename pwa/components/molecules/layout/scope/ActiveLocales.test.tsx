import ActiveLocales from './ActiveLocales'
import { renderWithProviders } from '~/services'
import catalog from '../../../../public/mocks/catalog.json'

describe('ActiveLocales', () => {
  it('match snapshot', () => {
    const { container } = renderWithProviders(
      <ActiveLocales content={catalog} />
    )
    expect(container).toMatchSnapshot()
  })
})
