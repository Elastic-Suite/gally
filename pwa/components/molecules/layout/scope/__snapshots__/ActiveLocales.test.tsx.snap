// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActiveLocales match snapshot 1`] = `
<div>
  <div
    class="css-18qndi"
  >
    <div
      class="css-1tdvoqh"
    >
      Total
    </div>
    <div
      class="css-1un2rp5"
    >
      <div
        class="css-13ehox3"
      >
        9
      </div>
      <div
        class="css-ry2e12"
      >
        activeLocale
      </div>
    </div>
    <div
      class="css-1sfqhqu"
    >
      <div>
        <div
          class="css-1tq682i"
        >
          TestUn (France)
        </div>
      </div>
      <div>
        <div
          class="css-1tq682i"
        >
          TestUnAA (France)
        </div>
      </div>
      <div>
        <div
          class="css-1tq682i"
        >
          TestAAUn (France)
        </div>
      </div>
      <div>
        <div
          class="css-1tq682i"
        >
          TAAestUn (France)
        </div>
      </div>
      <div>
        <div
          class="css-1tq682i"
        >
          TesAEFtUn (France)
        </div>
      </div>
      <div>
        <div
          class="css-1tq682i"
        >
          TesRDGFCDeux (France)
        </div>
      </div>
      <div>
        <div
          class="css-1tq682i"
        >
          DeutGFDsh (États-Unis)
        </div>
      </div>
      <div>
        <div
          class="css-1tq682i"
        >
          Français (France)
        </div>
      </div>
      <div>
        <div
          class="css-1tq682i"
        >
          Anglais (États-Unis)
        </div>
      </div>
    </div>
  </div>
</div>
`;
